File: CodeFestival/Q41-Q50/Q43_10진수를_2진수로.py
# 내가 푼 방법 : 재귀
n = int(input())
array = []
def to_binary(m, arr):
    arr.append(str(m % 2))
    m = m // 2
    if m == 0:
        arr.reverse()
        arr = ''.join(arr)
        return print(arr)
    else:
        to_binary(m, arr)

to_binary(n, array)

# 답안 : 반복문
# a = int(input())
# b = []

# while a:
#     print(a)
#     b.append(str(a % 2))
#     a = int(a / 2)

# print(b)
# b.reverse()
# print(b)
# print(''.join(b))

File: CodeFestival/Q41-Q50/Q43_10진수를_2진수로.js
let n = prompt("숫자를 입력하세요.");
const arr = [];
let result = '';
function to_binary(m, a) {
    a.push(m % 2);
    m = parseInt((m/2), 10);

    if (m == 0) {
        a.reverse();
        a.forEach((n) => {
            result += n;
        })
        return console.log(result);
    } else {
        to_binary(m, a);
    }
}

to_binary(n, arr)

// let a = prompt('10진수를 입력해주세요.')
// let b = [];
// let result = '';

// while (a){
// 	b.push(a % 2);
// 	a = parseInt(a / 2, 10);
// }
// b.reverse();

// b.forEach((n) => {
//   result += n;
// })

// console.log(result);